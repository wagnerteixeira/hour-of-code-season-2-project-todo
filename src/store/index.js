import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import layoutReducer from '../layout/redux/layoutReducer';
import todoReducer from '../todo/redux/todoReducer';

const reducers = combineReducers({  
  layout: layoutReducer,
  todo: todoReducer
});

export default createStore(reducers, applyMiddleware(thunk));
