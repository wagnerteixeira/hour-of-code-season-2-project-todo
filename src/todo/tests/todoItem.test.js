import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../store';

import TodoItem from '../todoItem';
import { todoMock } from './todoMock';

configure({ adapter: new Adapter() });

describe('Test <TodoItem />', function () {

    let component;
    let goTo = jest.fn();
    const _todoMock = { ...todoMock.data[0] }
    beforeEach(() => {
        component = shallow(
            <Provider store={store}>
                <TodoItem
                    goTo={goTo}
                    key={_todoMock.id}
                    todo={_todoMock}
                />
            </Provider>
        );
    });

    it('render <TodoItem /> ', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});