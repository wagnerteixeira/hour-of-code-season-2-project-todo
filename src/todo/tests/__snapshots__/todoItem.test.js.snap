// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoItem /> render <TodoItem />  1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(TodoItem)
    goTo={[MockFunction]}
    key="1"
    todo={
      Object {
        "assignee": "Assignee 1",
        "description": "Description 1",
        "id": "1",
        "status": "TODO",
        "title": "Title 1",
      }
    }
  />
</ContextProvider>
`;
